docs/docsite/rst/playbooks_tags.rst

```diff
@@ -86,27 +97,43 @@ You can apply tags to more than tasks, but they ONLY affect the tasks themselves
       tasks:
         ...
 
-You may also apply tags to roles::
+You may also apply tags to ``roles:``:
 
     roles:
       - { role: webserver, port: 5000, tags: [ 'web', 'foo' ] }
 
-And import statements::
+And to ``import_role:`` and ``import_tasks:`` statements::
+
+    - import_role:
+        name: myrole
+      tags: [web,foo]
 
     - import_tasks: foo.yml
       tags: [web,foo]
 
+
 All of these apply the specified tags to EACH task inside the play, imported
 file, or role, so that these tasks can be selectively run when the playbook
 is invoked with the corresponding tags.
 
-The above information does not apply to `include_tasks` or other dynamic includes,
-as the attributes applied to an include, only affect the include itself.
+Tags are inherited *down* the dependency chain. In order for tags to be
+applied to a role and all its dependencies, the tag should be applied to the
+role, not to all the tasks within a role.
+
+You can see which tags are applied to tasks and imported tasks by running
+``ansible-playbook`` with the ``--list-tasks`` option. You can display all
+tags applied to the tasks with the ``--list-tags`` option.
+
+.. note::
+    The above information does not apply to `include_tasks`, `include_roles`,
+    or other dynamic includes. Tags applied to either of these only tag the
+    include itself.
 
-Tags are inherited *down* the dependency chain. In order for tags to be applied to a role and all its dependencies,
-the tag should be applied to the role, not to all the tasks within a role.
+    To use tags with tasks and roles intended for use as dynamic `include`s,
+    all needed tasks should be explicitly tagged at the task level; or
+    ``block:`` may be used to tag more than one task at once. The include
+    itself should also be tagged.
```

Comment:
there is a distinction between `include` and the newer `include_tasks` and `include_role`, the former does push inheritance of tags to included tasks, the other 2 do not.

Reply:
In this note I just eliminated the use of 'include' as a proper term, and just used the concept 'dynamic inclusions'. In that same note block, 'include_tasks' and 'include_role' *is* called out specifically.

Reply:
fine with that, the module docs themselves should go into detail